src/utils.jl

```diff
@@ -24,6 +24,17 @@ function dot_matwise(x::AbstractArray, y::AbstractMatrix)
 end
 dot_matwise(x::AbstractMatrix, y::AbstractArray) = dot_matwise(y, x)
 
+function dot_vecwise(x::AbstractMatrix, y::AbstractMatrix)
+    return [dot(u, v) for (u, v) in zip(eachcol(x), eachcol(y))]
+end
+
+function dot_vecwise(x::AbstractMatrix, y::AbstractVector)
+    return [dot(u, y) for u in eachcol(x)]
+end
```

Comment:
I assume it is faster to compute
```suggestion
dot_vecwise(x::AbstractMatrix, y::AbstractVector) = x' * y
```